api/src/main/java/org/openmrs/util/LiquibaseVersionFinder.java

```diff
@@ -0,0 +1,148 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.util;
+
+import org.openmrs.module.VersionComparator;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class LiquibaseVersionFinder {
+
+	private static final String LIQUIBASE_SNAPSHOTS_FOLDER = "liquibase-snapshots";
+	private static final String LIQUIBASE_CORE_DATA_FILENAME = "liquibase-core-data.xml";
+	private static final String LIQUIBASE_SCHEMA_FILENAME = "liquibase-schema-only.xml";
+
+	private static final String LIQUIBASE_UPDATE_FOLDER = "liquibase-updates";
+	private static final String LIQUIBASE_UPDATE_FILENAME = "liquibase-update-to-latest.xml";
+
+	private File snapshotFolder;
+	private File updateFolder;
+	
+	public LiquibaseVersionFinder() {
+		snapshotFolder = new File(
+			this.getClass()
+				.getClassLoader()
+				.getResource( LIQUIBASE_SNAPSHOTS_FOLDER )
+				.getFile()
+		);
+
+		updateFolder = new File(
+			this.getClass()
+				.getClassLoader()
+				.getResource( LIQUIBASE_UPDATE_FOLDER )
+				.getFile()
+		);
+	}
+
+	public Set<List<String>> getLiquibaseChangesetCombinations() {
+		Set<List<String>> changesetCombinations = new HashSet<>();
+
+		for ( String snapshotVersion : getLiquibaseSnapshotVersions() ) {
+			List<String> changesetFilenames = new ArrayList<>();
+			
+			changesetFilenames.addAll( getLiquibaseSnapshotFilenames( snapshotVersion ) );
+			changesetFilenames.addAll( getApplicableLiquibaseUpdateFileNames( snapshotVersion ) );
+			
+			changesetCombinations.add( changesetFilenames );
+		}
+		
+		return changesetCombinations;
+	}
+
+	public Optional<String> getLatestLiquibaseSchemaSnapshotFilename() {
+		Optional<String> snapshotVersion = getLatestLiquibaseSnapshotVersion();
+		if ( snapshotVersion.isPresent() ) {
+			return Optional.of( LIQUIBASE_SNAPSHOTS_FOLDER + "/" + snapshotVersion.get() + "/" + LIQUIBASE_SCHEMA_FILENAME );
+		}
+		return Optional.empty();
+	}
+
+	public Optional<String> getLatestLiquibaseCoreDataSnapshotFilename() {
+		Optional<String> snapshotVersion = getLatestLiquibaseSnapshotVersion();
+		if ( snapshotVersion.isPresent() ) {
+			return Optional.of( LIQUIBASE_SNAPSHOTS_FOLDER + "/" + snapshotVersion.get() + "/" + LIQUIBASE_CORE_DATA_FILENAME );
```

Comment:
Although sometimes mere `/` work out of the box, just use the `File.separator` to cover all edge cases @WolfSchlegel